types/types.go

```diff
@@ -510,6 +510,11 @@ type SystemContext struct {
 	// === dir.Transport overrides ===
 	// DirForceCompress compresses the image layers if set to true
 	DirForceCompress bool
+
+	// CompressionFormat is the format to use for the compression of the blobs
+	CompressionFormat string
```

Comment:
Non-blocking: I somehow would like to have a dedicated type for that and constants for the compression algorithms to have a more idiomatic way of referencing them e.g., `if compressionFormat == compression.Zstd { ... }`.

Reply:
Instead of `== "zstd"`

Reply:
added a patch that defines constants for the various supported algorithms

Reply:
LGTM, thanks!